avm/res/security-insights/data-connector/main.bicep

```diff
@@ -0,0 +1,678 @@
+metadata name = 'Security Insights Data Connectors'
+metadata description = 'This module deploys a Security Insights Data Connector.'
+metadata owner = 'Azure/module-maintainers'
+
+import { roleAssignmentType } from 'br/public:avm/utl/types/avm-common-types:0.5.1'
+
+@description('Required. The resource ID of the Log Analytics workspace.')
+param workspaceResourceId string
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+// @allowed([
+//   'AmazonWebServicesCloudTrail'
+//   'AmazonWebServicesS3'
+//   'APIPolling'
+//   'AzureActiveDirectory'
+//   'AzureAdvancedThreatProtection'
+//   'AzureSecurityCenter'
+//   'Dynamics365'
+//   'GCP'
+//   'GenericUI'
+//   'IOT'
+//   'MicrosoftCloudAppSecurity'
+//   'MicrosoftDefenderAdvancedThreatProtection'
+//   'MicrosoftPurviewInformationProtection'
+//   'MicrosoftThreatIntelligence'
+//   'MicrosoftThreatProtection'
+//   'Office365'
+//   'Office365Project'
+//   'OfficeATP'
+//   'OfficeIRM'
+//   'OfficePowerBI'
+//   'PurviewAudit'
+//   'RestApiPoller'
+//   'ThreatIntelligence'
+//   'ThreatIntelligenceTaxii'
+// ])
+@description('Optional. The type of Data Connector.')
+param connectors dataConnectorType[]?
+
+#disable-next-line no-deployments-resources
+resource avmTelemetry 'Microsoft.Resources/deployments@2024-03-01' = if (enableTelemetry) {
+  name: '46d3xbcp.res.securityinsights-dataconnector.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+resource workspace 'Microsoft.OperationalInsights/workspaces@2022-10-01' existing = {
+  name: last(split(workspaceResourceId, '/'))
+}
+
+resource dataConnector 'Microsoft.SecurityInsights/dataConnectors@2024-10-01-preview' = [
```

Comment:
Happened to notice there's also a version `2025-03-01` available, if you want to use latest non-preview